src/main/java/duke/DukeParser.java

```diff
@@ -0,0 +1,85 @@
+package duke;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import duke.command.Command;
+import duke.command.CommandAddDeadline;
+import duke.command.CommandAddEvent;
+import duke.command.CommandAddTodo;
+import duke.command.CommandDelete;
+import duke.command.CommandDone;
+import duke.command.CommandExit;
+import duke.command.CommandHelp;
+import duke.command.CommandInvalid;
+import duke.command.CommandList;
+import task.TaskList;
+
+/**
+ * Parses any command that requires inputs
+ */
+public class DukeParser {
+
+    private final TaskList taskList;
+
+    /**
+     * Patterns for the Parser to look out for in the input
+     */
+    private final Pattern listPattern = Pattern.compile("list( .+)?", Pattern.CASE_INSENSITIVE);
```

Comment:
![image](https://user-images.githubusercontent.com/8487294/131513743-24d8d949-6d5f-4bfd-8323-1fd7c8631317.png)
Could be in UPPERCASE, and written like `PATTERN_LIST`, `PATTERN_DONE`, etc.

Reply:
No reason not to be static too.

Reply:
Ah I missed this out as well, thanks